lib/token.js

```diff
@@ -248,6 +249,131 @@ function addFragmentListener(sdk, windowEl, timeout) {
+    }
+
+    var clientId = oauthParams.clientId || sdk.options.clientId;
+    validateClaims(jwt.payload, sdk.options.url, clientId);
+
+    var idToken = {
+      idToken: res['id_token'],
+      claims: jwt.payload,
+      expiresAt: jwt.payload.exp,
+      scopes: scopes
+    };
+
+    if (Array.isArray(tokenTypes)) {
+      tokenDict['id_token'] = idToken;
+    } else {
+      return idToken;
+    }
+  }
+  
+  if (res['access_token']) {
+    var accessToken = {
+      accessToken: res['access_token'],
+      expiresAt: Number(res['expires_in']) + Math.floor(Date.now()/1000),
+      tokenType: res['token_type'],
+      scopes: scopes
+    };
+
+    if (Array.isArray(tokenTypes)) {
+      tokenDict['token'] = accessToken;
+    } else {
+      return accessToken;
+    }
+  }
+
+  if (!tokenDict['token'] && !tokenDict['id_token']) {
+    throw new AuthSdkError('Unable to parse OAuth flow response');
+  }
+
+  var tokens = [];
+
+  // Create token array in the order of the responseType array
+  for (var t = 0, tl = tokenTypes.length; t < tl; t++) {
+    var tokenType = tokenTypes[t];
+    if (tokenDict[tokenType]) {
+      tokens.push(tokenDict[tokenType]);
+    }
+  }
+
+  return tokens;
+}
+
+function getDefaultOAuthParams(sdk, oauthOptions) {
+  oauthOptions = util.clone(oauthOptions) || {};
+  var defaults = {
+    clientId: sdk.options.clientId,
+    redirectUri: sdk.options.redirectUri || window.location.href,
+    responseType: 'id_token',
+    responseMode: 'okta_post_message',
+    state: util.genRandomString(64),
+    nonce: util.genRandomString(64),
+    scope: ['openid', 'email']
+  };
+  util.extend(defaults, oauthOptions);
+  return defaults;
+}
+
+function convertOAuthParamsToQueryParams(oauthParams) {
+  // Quick validation
+  if (!oauthParams.clientId) {
+    throw new AuthSdkError('A clientId must be specified in the OktaAuth constructor to get a token');
+  }
+
+  if (util.isString(oauthParams.responseType) && oauthParams.responseType.indexOf(' ') !== -1) {
+    throw new AuthSdkError('Multiple OAuth responseTypes must be defined as an array');
+  }
+
+  // Convert our params to their actual OAuth equivalents
+  var oauthQueryParams = util.removeNils({
+    'client_id': oauthParams.clientId,
+    'redirect_uri': oauthParams.redirectUri,
+    'response_type': oauthParams.responseType,
+    'response_mode': oauthParams.responseMode,
+    'state': oauthParams.state,
+    'nonce': oauthParams.nonce,
+    'prompt': oauthParams.prompt,
+    'display': oauthParams.display,
+    'sessionToken': oauthParams.sessionToken,
+    'idp': oauthParams.idp
+  });
+
+  if (Array.isArray(oauthQueryParams['response_type'])) {
+    oauthQueryParams['response_type'] = oauthQueryParams['response_type'].join(' ');
+  }
+
+  if (oauthParams.scope.indexOf('openid') !== -1) {
+    oauthQueryParams.scope = oauthParams.scope.join(' ');
+  } else {
+    throw new AuthSdkError('openid scope must be specified in the scope argument');
```

Comment:
If I'm going down the accessToken route, do I need to specify openid as a scope?